nodes/Soroban/actions/settings/setOptions/helpers/helpers.ts

```diff
@@ -0,0 +1,44 @@
+import {
+	AuthClawbackEnabledFlag,
+	AuthRequiredFlag,
+	AuthRevocableFlag,
+	AuthImmutableFlag,
+	Signer,
+	AuthFlag,
+} from 'soroban-client';
+import IFlags from '../../../../../../common/interfaces/stellar/IFlags';
+import ISigner from '../../../../../../common/interfaces/stellar/ISigner';
+
+export function buildFlags(flags: IFlags['values']): AuthFlag {
+	const {
+		authorizationRequired,
+		authorizationRevocable,
+		authorizationInmutable,
+		authorizationClawbackEnabled,
+	} = flags;
+
+	let totalFlags = 0;
+
+	if (authorizationRequired) totalFlags += AuthRequiredFlag;
+	if (authorizationRevocable) totalFlags += AuthRevocableFlag;
+	if (authorizationInmutable) totalFlags += AuthImmutableFlag;
+	if (authorizationClawbackEnabled) totalFlags += AuthClawbackEnabledFlag;
+
+	return totalFlags as AuthFlag;
+}
+
+export function buildSigner(signerValues: ISigner['values']) {
+	const { signerType, signerKey, signerWeight } = signerValues;
+	const weight = Number(signerWeight);
+
+	if (typeof signerKey != 'string' && signerType != 'ed25519PublicKey') {
+		const signerData =
```

Comment:
If it's possible, avoid using the data word. Everything's data.
```suggestion
		const signer =
```